packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,383 @@
+    const hasher = createSHA256();
+    hasher.add(priorHash);
+    hasher.add(item);
+    return hasher.finish();
+  }
+
+  const sqlWriteTranscriptItem = db.prepare(`
+    INSERT INTO transcriptItems (vatID, item, position)
+    VALUES (?, ?, ?)
+    ON CONFLICT(vatID, position) DO UPDATE SET item = ?
+  `);
+
+  const sqlUpdateTranscriptSpan = db.prepare(`
+    UPDATE transcriptSpans
+    SET endPos = ?, hash = ?, size = ?
+    WHERE vatID = ? AND current = 1
+  `);
+
+  /**
+   * @param {string} vatID
+   * @param {string} item
+   */
+  const writeTranscriptItem = (vatID, item) => {
+    ensureTxn();
+    const { endPos, size, hash } = getCurrentTranscriptBounds(vatID);
+    !transcriptStatus.get(vatID) ||
+      Fail`can't write transcript ${q(vatID)} because it's already in use`;
+    item = `${item}\n`;
+    sqlWriteTranscriptItem.run(vatID, item, endPos, item);
+    sqlUpdateTranscriptSpan.run(
+      endPos + 1,
+      updateTranscriptHash(hash, item),
+      size + item.length,
+      vatID,
+    );
+  };
+
+  /**
+   * @param {string} name
+   * @param {SwingStoreExporter} exporter
+   * @param {Map} metadata
+   * @returns {Promise<void>}
+   */
+  async function importTranscript(name, exporter, metadata) {
+    const info = JSON.parse(metadata.get(name));
+    const parts = name.split('.');
+    const [tag, type, vatID, rawStartPos] = parts;
+    assert(
+      parts.length === 4 && tag === 'export' && type === 'transcript',
+      `expected transcript name of the form 'export.transcript.{vatID}.{startPos}', saw '${name}'`,
+    );
+    assert(
+      info.vatID === vatID,
+      `transcript name says vatID ${vatID}, metadata says ${info.vatID}`,
+    );
+    const startPos = Number(rawStartPos);
+    assert(
+      info.startPos === startPos,
+      `transcript name says startPos ${startPos}, metadata says ${info.startPos}`,
+    );
+    const artifactChunks = exporter.getArtifact(name);
+    const inStream = Readable.from(artifactChunks);
+    const lineTransform = new ReadlineTransform();
+    const lineStream = inStream.pipe(lineTransform);
+    let hash = initialHash;
+    let pos = startPos;
+    let totalSize = 0;
+    for await (const line of lineStream) {
+      const item = `${line}\n`;
+      sqlWriteTranscriptItem.run(vatID, item, pos, item);
+      hash = updateTranscriptHash(hash, item);
+      pos += 1;
+      totalSize += item.length;
+    }
+    assert(
+      info.size === totalSize,
+      `transcript ${name} size is ${totalSize}, metadata says ${info.size}`,
+    );
+    assert(
+      info.hash === hash,
+      `transcript ${name} hash is ${hash}, metadata says ${info.hash}`,
+    );
+    sqlWriteSpan.run(
+      info.vatID,
+      info.startPos,
+      info.endPos,
+      info.hash,
+      info.current,
+      info.size,
+    );
+  }
+
+  /** @param {string} vatID */
+  const closeTranscript = vatID => {
+    transcriptStatus.delete(vatID);
+  };
+
+  return harden({
```

Comment:
Talking with @FUDCo , he's keen on `Pos` and `Item` for their brevity. I think we settled on a hierarchy of:
* "transcript": one per vatID
* "span" many per transcript, each with a unique startPos
* "item": many per span

The API method names should be VERB+NAME, where NAME is one of the above three. I think I also talked him into `addItem` for the API that appends a new item to the current span. I proposed `rotateSpan` to evoke a parallel with rotating your logfiles.

Reply:
I've done a top-to-bottom rationalization/cleanup pass on the names of everything in `transcriptStore.js`.  Let me know what you think once I push the next update.

Reply:
I feel pretty strongly that "rollover" is exactly the right verb for the operation that is being performed.  On the other hand, perhaps `rolloverSpan` is not quite so ideal, as the object of the verb is kind of funny. I was trying to follow @warner's transcript/span/item taxonomy strictly, but there's an argument to be made that what's being rolled over is the transcript itself rather than the span, since the operation spans two spans (so to speak).  I'd be happy with `rolloverTranscript` or even `rolloverTranscriptSpan` as alternatives (or possibly even `rolloverTransciptSpans`, though that would leave us vulnerable to "was this that one call that was plural?" kind of confusion)